src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/AndroidGradleProjectTests.cs

```diff
@@ -209,6 +209,33 @@ public void BuildIncremental ()
 			StringAssert.Contains (@"uses-sdk android:minSdkVersion=""30""", packagedManifestContent);
 		}
 
+		[Test]
+		public void BuildCustomOutputPaths ()
+		{
+			var gradleProject = AndroidGradleProject.CreateDefault (GradleTestProjectDir);
+			var moduleName = gradleProject.Modules.First ().Name;
+
+			using var builder = CreateDllBuilder ();
+			var customOutputPathsRoot = Path.Combine (Root, builder.ProjectDirectory, "customout");
+			var gradleOutputPath = Path.Combine (customOutputPathsRoot, "gradleoutdir");
+
+			var proj = new XamarinAndroidLibraryProject {
+				OtherBuildItems = {
+					new BuildItem (KnownProperties.AndroidGradleProject, gradleProject.BuildFilePath) {
+						Metadata = {
+							{ "ModuleName", moduleName },
+							{ "OutputPath", gradleOutputPath },
+						},
+					},
+				},
+				OutputPath = Path.Combine (customOutputPathsRoot, "outdir"),
+				IntermediateOutputPath = Path.Combine (customOutputPathsRoot, "intermediatedir"),
```

Comment:
Should we also test `$(UseArtifactsOutput)`?

* https://learn.microsoft.com/dotnet/core/sdk/artifacts-output

This is probably a common way people would move the folders around now, likely more and more.

Reply:
Thanks good idea, I've added another `BuildArtifactsOutputPaths` test to try to cover this.